src/simplewallet/simplewallet.cpp

```diff
@@ -2372,7 +2374,19 @@ bool simple_wallet::transfer_main(int transfer_type, const std::vector<std::stri
     }
   }
 
-  const size_t min_args = (transfer_type == TransferLocked) ? 3 : 2;
+  size_t min_args;
+  switch(transfer_type)
+  {
+    case TransferStake:
+      min_args = 5;
+    break;
+    case TransferLocked:
+      min_args = 3;
+    break;
+    default:
+      min_args = 2;
+    break;
+  }
```

Comment:
A switch statement here seems like a massive overkill and a big drop in readability compared to a double ternary:
```C++
const size_t min_args = (transfer_type == TransferLocked) ? 3 :
                        (transfer_type == TransferStake ) ? 5 :
                        2;
```